datasetinsights/io/gcs.py

```diff
@@ -1,25 +1,57 @@
+import base64
 import logging
 import os
+from os import makedirs
+from os.path import basename, isdir
 from pathlib import Path
 
 from google.cloud.storage import Client
 
+from datasetinsights.io.download import validate_checksum
+from datasetinsights.io.exceptions import ChecksumError, DownloadError
+
 logger = logging.getLogger(__name__)
 
 
 class GCSClient:
+    """ This class is used to download data from GCS location
+        and perform function such as downloading the dataset and checksum
+        validation.
+    """
+
     def __init__(self, **kwargs):
         """ Initialize a client to google cloud storage (GCS).
         """
         self.client = Client(**kwargs)
 
-    def download(self, bucket_name, object_key, localfile):
-        """ Download a single object from GCS
+    def download(self, *, url=None, local_path=None, bucket=None, key=None):
```

Comment:
Validate that the key and bucket are not None if url is not provided.

Reply:
If bucket is non GCS API will throw error